exporter/sumologicexporter/exporter.go

```diff
@@ -16,44 +16,113 @@ package sumologicexporter
 
 import (
 	"context"
+	"net/http"
+
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer/pdata"
-	"go.uber.org/zap"
+)
+
+const (
+	logKey        string = "log"
+	maxBufferSize int    = 100
 )
 
 type sumologicexporter struct {
-	logger *zap.Logger
+	config *Config
+	client *http.Client
+	filter *filtering
 }
 
-var _ (component.LogsExporter) = (*sumologicexporter)(nil)
-
 func newLogsExporter(
-	logger *zap.Logger,
+	cfg *Config,
 ) (component.LogsExporter, error) {
-	se := &sumologicexporter{logger: logger}
-	return se, nil
+	se, err := initExporter(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return exporterhelper.NewLogsExporter(
+		cfg,
+		se.pushLogsData,
+		// Disable exporterhelper Timeout, since we are using a custom mechanism
+		// within exporter itself
+		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0}),
+		exporterhelper.WithRetry(cfg.RetrySettings),
+		exporterhelper.WithQueue(cfg.QueueSettings),
+	)
 }
 
-func newMetricsExporter(
-	logger *zap.Logger,
-) (component.MetricsExporter, error) {
-	se := &sumologicexporter{logger: logger}
+func initExporter(cfg *Config) (*sumologicexporter, error) {
+	filter, err := newFiltering(cfg.MetadataAttributes)
+	if err != nil {
+		return nil, err
+	}
+
+	se := &sumologicexporter{
+		config: cfg,
+		client: &http.Client{
+			Timeout: cfg.TimeoutSettings.Timeout,
+		},
+		filter: filter,
+	}
+
 	return se, nil
 }
 
-func (se *sumologicexporter) Start(_ context.Context, _ component.Host) error {
-	return nil
-}
+// pushLogsData groups data with common metadata and send them together to Sumo Logic
+func (se *sumologicexporter) pushLogsData(ctx context.Context, ld pdata.Logs) (droppedTimeSeries int, err error) {
```

Comment:
droppedTimeSeries in the context of logs? hmm

Reply:
I am using official function signature: https://github.com/open-telemetry/opentelemetry-collector/blob/master/exporter/exporterhelper/logshelper.go#L29-L31